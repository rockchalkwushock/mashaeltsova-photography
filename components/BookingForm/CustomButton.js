import { styles } from '../../lib'

const Button = ({ children, valid }) => (
  <button className="form-button" disabled={valid} type="submit">
    {children}
    <style jsx>{`
      .form-button {
        background-color: ${styles.colors.peach};
        border: none;
        color: ${styles.colors.warm};
        font-size: 1.25em;
        font-family: ${styles.fonts.nunito};
        font-size: 1em;
        grid-column: span 6;
        padding: 0.5em;
        text-align: center;
        width: 80%;
      }
    `}</style>
  </button>
)

export default Button
