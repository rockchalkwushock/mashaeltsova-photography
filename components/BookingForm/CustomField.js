import CustomInput from './CustomInput'
import CustomLabel from './CustomLabel'
import CustomSpan from './CustomSpan'
import CustomTextArea from './CustomTextArea'

const CustomField = ({ label, input, type, meta: { error, touched } }) => {
  if (input.name === 'message') {
    return (
      <div className="field">
        <CustomLabel label={label} />
        <CustomTextArea input={input} placeholder={label} />
        {touched && (error && <CustomSpan>{error}</CustomSpan>)}
        <style>{`
      .field {
        grid-column: span 6;
        text-align: center;
        width: 100%;
      }
    `}</style>
      </div>
    )
  }
  return (
    <div className="field">
      <CustomLabel label={label} />
      <CustomInput input={input} placeholder={label} type={type} />

      {touched && (error && <CustomSpan>{error}</CustomSpan>)}
      <style>{`
      .field {
        grid-column: span 6;
        text-align: center;
        width: 100%;
      }
    `}</style>
    </div>
  )
}

export default CustomField
