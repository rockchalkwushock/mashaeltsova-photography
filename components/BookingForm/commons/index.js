import Button from './CustomButton'
import Form from './CustomForm'
import Input from './CustomInputField'
import TextArea from './CustomTextAreaField'

export { Button, Form, Input, TextArea }
