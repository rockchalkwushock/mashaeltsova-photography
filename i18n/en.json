{
  "buttonBooking": "Book Now!",
  "buttonGallery": "View my work here!",
  "formEmail": "Email",
  "formFirstName": "First Name",
  "formLastName": "Last Name",
  "formMessage": "Short Message",
  "formPhone": "Phone",
  "formErrorEmailPattern": "Not a valid email.",
  "formErrorNamePattern":
    "Not a valid pattern. May only contain upper and lowercase letters.",
  "formErrorMessagePattern":
    "May only contain upper & lower case letters and numbers.",
  "formErrorPhonePattern":
    "Should be match one of the following: +11231231234, 11231231234, 1231231234,",
  "formErrorRequired": "Please fill this in.",
  "formErrorTooLong": "Max length 140 characters.",
  "formErrorTooShort": "Minimum length 2 characters.",
  "formRequiredMessage": "All fields required.",
  "galleryNav": "Family,Portrait,Travel,Wedding",
  "modalError": "I'm sorry there seems to be a problem, please try back later.",
  "modalLoading": "Your request is being sent now.",
  "modalSuccess": "Your booking inquiry has been sent!",
  "sectionAbout":
    "For many years now I have had a great passion for capturing a moment in time and sharing the beauty of that moment with the world. I enjoy taking pictures for many varying events. When I am not doing photo shoots you will find me preforming English tutoring, or practicing my other passion, surfing.",
  "sectionPlug":
    "Contact me if you want to relive those precious moments for years to come!",
  "sectionServices":
    "I am willing to negotiate other requested services, however the below are what I primarily offer: Baby Portraits,Family Portraits,Self-Portraits,Engagements,Weddings",
  "sideNav": "Home,Gallery"
}
