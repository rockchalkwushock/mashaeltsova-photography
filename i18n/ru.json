{
  "buttonBooking": "Забронировать сейчас!",
  "buttonGallery": "Мои работы можно посмотреть тут!",
  "formEmail": "Электронный адрес",
  "formFirstName": "Имя",
  "formLastName": "Фамилия",
  "formMessage": "Сообщение",
  "formPhone": "Телефон",
  "formErrorEmailPattern": "FIXME",
  "formErrorNamePattern": "FIXME",
  "formErrorMessagePattern":
    "Может содержать заглавные и строчные буквы, цифры, и .!?,;'",
  "formErrorPhonePattern":
    "Должно соответствовать одному из следующих: +1-234-567-8901, +61-234-567-89-01 +46-234 5678901",
  "formErrorRequired": "Пожалуйста, заполните это поле.",
  "formErrorTooLong": "Максимальная длина 140 знаков.",
  "formErrorTooShort": "минимальная длина 2 знака.",
  "formRequiredMessage": "Необходимо заполнить все поля.",
  "galleryNav": "Семья,Портрет,Путешествия,Свадьбы",
  "modalError": "Произошла ошибка, попробуйте еще раз.",
  "modalLoading": "Ваш запрос отправляется.",
  "modalSuccess": "Ваш запрос на фотосессию отправлен!",
  "sectionAbout":
    "Уже много лет я занимаюсь фотографией и мне нравится моя работа. Мне нравится общаться и фотографировать людей. Если я не фотографирую то, путешествую или учу английский язык, или просто наслаждаюсь жизнью.",
  "sectionPlug": "Связитесь со мной , если хотите обсудить съемку!",
  "sectionServices":
    "Открыта к обсуждению съемок, специализируюсь на следующих съемках: Детская и семейная съемка,Портреты,Помолвки,Свадьбы",
  "sideNav": "FIXME"
}
