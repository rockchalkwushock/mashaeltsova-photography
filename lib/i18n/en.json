{
  "description": "Photography Business Website for Masha Eltsova Photography",
  "greeting": "Hello World!",
  "title": "Masha Eltsova Photography"
}