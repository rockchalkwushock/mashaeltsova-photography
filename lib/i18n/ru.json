{
  "book": "Забронируйте сейчас!",
  "description": "Сайт фотографии для Маши Ельцовой",
  "email": "Эл. адрес",
  "emailInvalid": "Неверный адрес электронной почты.",
  "emailRequired": "Требуется электронная почта.",
  "errorResponse":
    "У вас возникла проблема с вашим запросом, пожалуйста, попробуйте позже.",
  "firstName": "Имя",
  "firstNameRequired": "Имя требуется.",
  "firstNameInvalid": "Неверное имя. Только альфа-символы.",
  "firstNameLength": "Имя должно быть не менее 2 символов.",
  "greeting": "Привет мир!",
  "lastName": "Фамилия",
  "lastNameInvalid": "Недопустимое имя. Только альфа-символы.",
  "lastNameLength": "Фамилия должна быть не менее 2 символов.",
  "lastNameRequired": "Фамилия обязательна.",
  "message": "Каковы ваши потребности в бронировании?",
  "phone": "Номер телефона",
  "phoneInvalid":
    "Неправильный номер телефона. Принимает: +1-234-567-8901, +61-234-567-89-01.",
  "phoneRequired": "Требуется номер телефона.",
  "successResponse": "Ваш запрос на бронирование был отправлен!",
  "title": "Маша Ельцова Фотография",
  "messageRequired": "Сообщение требуется.",
  "messageInvalid": "FIXME",
  "messageMaxLength": "Сообщение должно быть не более 140 символов.",
  "messageMinLength": "Сообщение должно содержать не менее 10 символов.",
  "rcws": "Разработка RCWS"
}
