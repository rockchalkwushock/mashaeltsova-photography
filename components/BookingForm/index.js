/* eslint-disable no-nested-ternary */
import React, { Component } from 'react'
import { Field, reduxForm } from 'redux-form'
import PropType from 'prop-types'

import { messages } from '../../lib'
import validate from './validate'
import CustomButton from './CustomButton'
import CustomField from './CustomField'
import CustomSpan from './CustomSpan'
import Form from './Form'

/**
 * REVIEW
 *
 * I would like to when an input is correct display a green circle with a white
 * check mark to the user or red circle with white 'X'. I think it would be
 * really cool if when the user moved to the next input:
 *  'valid': icon rolled out from behind input and border highlight green.
 *  'invalid': icon rolled out from behind input and border highlight red as well as
 *            display error message. Shake input too.
 *
 * For time being this.renderError() & this.renderSuccess() will do.
 * In the future on:
 *  'error': overlay a modal for X seconds that says there was an internal error.
 *  'success': transition form to a successful icon or message using animations.
 */

class BookingForm extends Component {
  static propTypes = {
    intl: PropType.object.isRequired
  }
  state = {
    error: false,
    isFetched: false
  }
  submitForm = async values => {
    const { getBookingInfo, reset } = this.props
    return getBookingInfo(values).then(({ payload }) => {
      // NOTE: Checking that this is a successful request.
      // No need to set the message from the server on state
      // use `react-intl` for translations.
      if (payload.message) {
        this.setState(state => ({
          ...state,
          isFetched: true
        }))
      } else if (payload.error) {
        this.setState(state => ({
          ...state,
          error: true,
          isFetched: true
        }))
      }
      reset()
    })
  }
  renderError = () => (
    <CustomSpan>{this.props.intl.formatMessage(messages.errorRes)}</CustomSpan>
  )
  renderForm = () => {
    const { handleSubmit, intl, valid } = this.props
    return (
      <Form onSubmit={handleSubmit(this.submitForm)}>
        <Field
          component={CustomField}
          label={intl.formatMessage(messages.firstName)}
          name="firstName"
          type="text"
        />
        <Field
          component={CustomField}
          label={intl.formatMessage(messages.lastName)}
          name="lastName"
          type="text"
        />
        <Field
          component={CustomField}
          label={intl.formatMessage(messages.email)}
          name="email"
          type="email"
        />
        <Field
          component={CustomField}
          label={intl.formatMessage(messages.phone)}
          name="phone"
          type="tel"
        />
        <Field
          component={CustomField}
          label={intl.formatMessage(messages.message)}
          name="message"
        />
        <CustomButton valid={!valid}>
          {intl.formatMessage(messages.book)}
        </CustomButton>
      </Form>
    )
  }
  renderSuccess = () => (
    <CustomSpan>
      {this.props.intl.formatMessage(messages.successRes)}
    </CustomSpan>
  )
  render() {
    const { error, isFetched } = this.state
    const view =
      !isFetched && !error
        ? this.renderForm()
        : isFetched && error ? this.renderError() : this.renderSuccess()
    return view
  }
}

export default reduxForm({
  form: 'booking',
  validate
})(BookingForm)
