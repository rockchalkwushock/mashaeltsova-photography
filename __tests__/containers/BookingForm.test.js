import React from 'react'
import { render } from 'enzyme'

import BookingForm from '../../containers/FormContainer'
import en from '../../i18n/en.json'
import ru from '../../i18n/ru.json'

describe('Component: BookingForm', () => {
  test('Renders without exploding in English', () => {
    const tree = render(<BookingForm locale="en" messages={en} />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding in Russian', () => {
    const tree = render(<BookingForm locale="ru" messages={ru} />)
    expect(tree).toMatchSnapshot()
  })
})
