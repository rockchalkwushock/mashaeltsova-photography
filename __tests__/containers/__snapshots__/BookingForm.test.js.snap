// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BookingForm Renders without exploding in English 1`] = `
<form
  class="jsx-1188712385"
>
  <div
    class="jsx-2142180042 firstName"
  >
    <label
      class="jsx-2142180042"
      for="firstName"
    >
      First Name
    </label>
    <input
      class="jsx-2142180042"
      name="firstName"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 lastName"
  >
    <label
      class="jsx-2142180042"
      for="lastName"
    >
      Last Name
    </label>
    <input
      class="jsx-2142180042"
      name="lastName"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 email"
  >
    <label
      class="jsx-2142180042"
      for="email"
    >
      Email
    </label>
    <input
      class="jsx-2142180042"
      name="email"
      type="email"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 phone"
  >
    <label
      class="jsx-2142180042"
      for="phone"
    >
      Phone
    </label>
    <input
      class="jsx-2142180042"
      name="phone"
      type="tel"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 message"
  >
    <label
      class="jsx-2142180042"
      for="message"
    >
      Short Message
    </label>
    <input
      class="jsx-2142180042"
      name="message"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-601574895 button"
  >
    <button
      class="jsx-601574895"
      type="submit"
    >
      Book Now!
    </button>
  </div>
</form>
`;

exports[`Component: BookingForm Renders without exploding in Russian 1`] = `
<form
  class="jsx-1188712385"
>
  <div
    class="jsx-2142180042 firstName"
  >
    <label
      class="jsx-2142180042"
      for="firstName"
    >
      Имя
    </label>
    <input
      class="jsx-2142180042"
      name="firstName"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 lastName"
  >
    <label
      class="jsx-2142180042"
      for="lastName"
    >
      Фамилия
    </label>
    <input
      class="jsx-2142180042"
      name="lastName"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 email"
  >
    <label
      class="jsx-2142180042"
      for="email"
    >
      Электронный адрес
    </label>
    <input
      class="jsx-2142180042"
      name="email"
      type="email"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 phone"
  >
    <label
      class="jsx-2142180042"
      for="phone"
    >
      Телефон
    </label>
    <input
      class="jsx-2142180042"
      name="phone"
      type="tel"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-2142180042 message"
  >
    <label
      class="jsx-2142180042"
      for="message"
    >
      Сообщение
    </label>
    <input
      class="jsx-2142180042"
      name="message"
      type="text"
      value=""
    />
    <span
      class="jsx-2142180042 errors"
    />
  </div>
  <div
    class="jsx-601574895 button"
  >
    <button
      class="jsx-601574895"
      type="submit"
    >
      Забронировать сейчас!
    </button>
  </div>
</form>
`;
